core/domain/stats_domain.py

```diff
@@ -41,6 +41,161 @@
+                    self.num_completions))
+
+        if not isinstance(self.state_stats_mapping, dict):
+            raise utils.ValidationError(
+                'Expected state_stats_mapping to be a dict, received %s' % (
+                    self.state_stats_mapping))
+
+
+class StateStats(object):
+    """Domain object representing analytics data for an exploration's state."""
+
+    def __init__(
+            self, total_answers_count, useful_feedback_count,
+            learners_answered_correctly, total_hit_count, first_hit_count,
+            total_solutions_triggered_count):
+        """Constructs a StateStats domain object.
+
+        Args:
+            total_answers_count: int. Total number of answers submitted to this
+                state.
+            useful_feedback_count: int. Total number of answers that received
+                useful feedback.
+            learners_answered_correctly: int. Number of learners who submitted
+                correct answers to the state.
+            total_hit_count: int. Total number of times the state was entered.
+            first_hit_count: int. Number of times the state was entered for the
+                first time.
+            total_solutions_triggered_count: int. Number of times the solution
+                button was triggered to answer a state.
+        """
+        self.total_answers_count = total_answers_count
+        self.useful_feedback_count = useful_feedback_count
+        self.learners_answered_correctly = learners_answered_correctly
+        self.total_hit_count = total_hit_count
+        self.first_hit_count = first_hit_count
+        self.total_solutions_triggered_count = total_solutions_triggered_count
+
+    def to_dict(self):
+        """Returns a dict representation of the domain oject."""
+        state_stats_dict = {
+            'total_answers_count': self.total_answers_count,
+            'useful_feedback_count': self.useful_feedback_count,
+            'learners_answered_correctly': self.learners_answered_correctly,
+            'total_hit_count': self.total_hit_count,
+            'first_hit_count': self.first_hit_count,
+            'total_solutions_triggered_count': (
+                self.total_solutions_triggered_count)
+        }
+        return state_stats_dict
+
+    @classmethod
+    def from_dict(cls, state_stats_dict):
+        """Constructs a StateStats domain object from a dict."""
+        return cls(
+            state_stats_dict['total_answers_count'],
+            state_stats_dict['useful_feedback_count'],
+            state_stats_dict['learners_answered_correctly'],
+            state_stats_dict['total_hit_count'],
+            state_stats_dict['first_hit_count'],
+            state_stats_dict['total_solutions_triggered_count']
+        )
+
+    def validate(self):
+        """Validates the StateStats domain object."""
+
+        if not isinstance(self.total_answers_count, int):
```

Comment:
Want to validate >= 0 as well? (Basically, try to validate anything that can be validated.)

Reply:
Done. Modified the structure of the validate() function as well.